consumer_metadata_response.go

```diff
@@ -20,10 +20,14 @@ func (r *ConsumerMetadataResponse) decode(pd packetDecoder) (err error) {
 	}
 	r.Err = KError(tmp)
 
-	r.Coordinator = new(Broker)
-	if err := r.Coordinator.decode(pd); err != nil {
+	coordinator := new(Broker)
+	if err := coordinator.decode(pd); err != nil {
 		return err
 	}
+	if coordinator.addr == ":0" {
```

Comment:
It is necessary to make encode/decode symmetric. When a response that contains zero coordinator info is encoded it used to contain a zero instance of broker, but a response object that contains no broker had a `nil` broker reference.